packages/filesystem/src/node/filesystem-watcher-client.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as path from "path";
+import { injectable, inject } from "inversify";
+import { JsonRpcProxyFactory, ILogger, ConnectionErrorHandler, DisposableCollection, Disposable } from "@theia/core";
+import { IPCConnectionProvider } from "@theia/core/lib/node/messaging";
+import { FileSystemWatcherServer, WatchOptions, FileSystemWatcherClient, ReconnectingFileSystemWatcherServer } from "../common/filesystem-watcher-protocol";
+
+export const CHOKIDAR_WATCHER = 'chokidar-watcher';
+
+@injectable()
+export class FileSystemWatcherServerClient implements FileSystemWatcherServer {
+
+    protected readonly proxyFactory = new JsonRpcProxyFactory<FileSystemWatcherServer>();
+    protected readonly remote = new ReconnectingFileSystemWatcherServer(this.proxyFactory.createProxy());
```

Comment:
Just for my understanding is it better to do these new inside the constructor or outside or it doesn't matter ?

I've seen @kittaakos move some to inside a constructor and I tought maybe it made sense to allocate the memory only on contruction... but I'm not sure if it really matters.

Should we have a policy on that?

Reply:
I don't think it is matter and we don't have a policy. For me, it looks less verbose, since I don't need to declare the type.